src/main/php/PDepend/Util/Utf8Util.php

```diff
@@ -66,14 +68,51 @@ public static function ensureEncoding($raw)
         }
 
         $text = '';
+
+        // Try to convert raw text to UTF8 using "iconv", if exists.
         if (function_exists('iconv')) {
             $text = @iconv($encoding, 'UTF8//IGNORE', $raw) ?: '';
         }
 
-        if ($text === '') {
+        // Then try with "mbstring" extension, if available.
+        if ($text === '' && extension_loaded('mbstring')) {
+            $text = mb_convert_encoding($raw, 'UTF-8', mb_list_encodings());
+        }
+
+        // Then try with native PHP function, if not deprecated.
+        if ($text === '' && PHP_VERSION_ID < 80200) {
             $text = utf8_encode($raw);
         }
```

Comment:
```suggestion
        // Then try with native PHP function, if not removed.
        if ($text === '' && function_exists('utf8_encode')) {
            $text = @utf8_encode($raw);
        }
```
I'd say, in practice using the native method or the polyfill is the same except than native function will be faster. So better use the native one as long as it's not removed.

Reply:
Ok, fine with me. I usually not a fan of using the `@` error suppressor token, but in term of performance, you're right.

Thanks!